src/util/analytics-client.ts

```diff
@@ -0,0 +1,92 @@
+import { getLogger 	} from "config/logger";
+import { omit } from "lodash";
+import { optionalRequire } from "optional-require";
+import { isNodeProd } from "utils/is-node-env";
+
+const { analyticsClient } = optionalRequire("@atlassiansox/analytics-node-client") || {};
+const logger = getLogger("analytics");
+const instance = process.env.INSTANCE_NAME;
+const appKey = `com.github.integration${instance ? `.${instance}` : ""}`;
+
+let analyticsNodeClient;
+
+export function sendAnalytics(eventType: "trait")
+export function sendAnalytics(eventType: "screen", attributes:{name:string} & Record<string,unknown>)
+export function sendAnalytics(eventType: "ui" | "track" | "operational", attributes: Record<string,unknown>)
+export function sendAnalytics(eventType: string, attributes?: Record<string, unknown>): void {
+	if (!analyticsClient && !isNodeProd()){
+		return;
+	}
+
+	if (!analyticsNodeClient){
+		// Values defined by DataPortal. Do not change their values as it will affect our metrics logs and dashboards.
+		analyticsNodeClient = analyticsClient({
+			env: "prod", // This needs to be "prod" as we're using prod Jira instances.
+			product: "gitHubForJira"
+		});
+	}
+
+	const baseAttributes = {
+		userId: "anonymousId",
+		userIdType: "atlassianAccount",
+		tenantIdType: "cloudId",
+		tenantId: "NONE"
+	};
+
+	attributes!.appKey = appKey;
+
+	logger.debug({ eventType }, "Sending analytics");
+
+	switch (eventType) {
+		case "screen":
+			wrapPromise(analyticsNodeClient.sendScreenEvent({
+				...baseAttributes,
+				name: attributes?.name,
+				screenEvent: {
+					platform: "web",
+					attributes: omit(attributes, "name")
+				}
+			}));
+			break;
+		case "ui":
+			wrapPromise(analyticsNodeClient.sendUIEvent({
+				...baseAttributes,
+				uiEvent: {
+					attributes
+				}
+			}));
+			break;
+		case "track":
+			wrapPromise(analyticsNodeClient.sendTrackEvent({
+				...baseAttributes,
+				trackEvent: {
+					attributes
+				}
+			}));
+			break;
+		case "trait":
+			wrapPromise(analyticsNodeClient.sendTraitEvent({
+				entityType: attributes?.entityType,
+				entityId: attributes?.entityId,
+				entityTraits: attributes?.entityTraits
+			}));
+			break;
+		case "operational":
+			wrapPromise(analyticsNodeClient.sendOperationalEvent({
+				...baseAttributes,
+				operationalEvent: {
+					attributes
+				}
+			}));
+			break;
+		default:
+			logger.warn(`Cannot sendAnalytics: unknown eventType`);
+			break;
+	}
+}
+
+function wrapPromise(promise: Promise<unknown>) {
```

Comment:
The analytics client was created by @atrigueiro and merged back in March https://github.com/atlassian/github-for-jira/pull/1093 but was reverted because it was causing builds to fail. I've just copied her code over and fixed the build issues. Happy to make changes we think are really important but would otherwise prefer to leave as is. 

That said, I would be happy to potentially change the name of this function but don't think`logWarnOnError` makes sense. The calls to `wrapPromise` send analytics e.g.
```
wrapPromise(analyticsNodeClient.sendTraitEvent({
				entityType: attributes?.entityType,
				entityId: attributes?.entityId,
				entityTraits: attributes?.entityTraits
			}));
```
It falls into the catch block and sends a warning message if something goes wrong but this isn't the purpose of the function. Something like `sendEvent` or `sendAnalyticEvent` are possible options.